### PR TITLE
Redesign of install process and screens for improved branding and consistency.

### PR DESCRIPTION
**Pull request by johnalarcon**
This PR pertains to issue #210 and implements a redesign of the installation process and views.
It adds consistency and brand recognition to the entire process and cleans up the related code.

## Description
The goals achieved with this PR are several:
1. create a faster install process, akin to an "Amazing 60 Second Install" (or something)
2. replace coin-style logo with wordmark logo
3. brand primary install screens with logo, CP color, and CTA-style buttons
4. brand secondary (error) screens with logo and secondary-style buttons (or links, as appropriate)
5. remove verbosity from texts; link to help-docs to avoid lengthy in-place descriptions
6. improve font sizes
7. remove needless steps
8. cascade in CSS rules for button colors; avoid changing rules affecting entire admin

### About the Install Process
This update breaks the install process into several main views, each of which have a CTA-stlye button:
1. Choose language
2. Setup database
3. Setup admin
4. Enjoy success

There are also quite a few supporting views for the many various errors that might occur during installation. These are handled and branded, as well.

### About the Logo File
I've added png/svg logo files to `./wp-admin/images/` as they're install-related - perhaps these should be moved to `./wp-includes/images/` where there are other brand assets. The filenames should be reviewed and a naming schema decided.
	
### About the Texts
Most of the install and error texts have been rewritten. The goal was to remove verbosity, yet remain informative. I feel a nice balance has been struck, but am open to suggestions as few of these have been discussed. A notable exception is in the case of database errors - this markup is provided to the install process, not generated within it, so it is used as-is.
	
### Additional
Probably the biggest change code-wise is the removal of the on/off PHP tags. My syntax highlighting was ineffective with it and my IDE's code collapse was "dancing"... I hope this is not a problem. In some places, I've outputted tabs and newlines to make the source markup readable; these aren't really needed and can be removed.

### Bonus
The "Alright Sparky" step is removed in support of this [petition](https://petitions.classicpress.net/posts/95/remove-the-alright-sparky-step-during-installation). This petition has decent support, but does not reach the 50 vote threshold. As we are pre-1.0, perhaps a committee decision can be made to allow it. This step can be added back in, if absolutely necessary.

## How has this been tested?
Manual, localhost, painstakingly. That said, it will be good to get a bunch of eyes on it.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-breaking enhancement.

## Checklist:
- [x] My code follows the code style of this project.
- [x] It is not clear if this change will require documentation updates.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### Primary Views

**Choose Language**
main-choose-language

**Database Setup**
main-database-setup

**Admin Setup**
main-admin-setup

**Install Success** when user chose own password
main-success-1

**Install Success** when system chose the password
main-success-2

### Secondary Views

**Warning** config file exists in install
warning-config-file-found-1

**Warning** config file exists 1 level up from install
warning-config-file-found-2

**Error** selecting database
error-database-selection

**Error** connecting to database
error-database-connection

**Error** missing table prefix
error-table-prefix-missing

**Error** invalid table prefix
error-table-prefix-invalid

**Error** invalid username
error-admin-username

**Error** invalid email
error-admin-email

**Error** insufficient MySQL version
error-mysql-insufficient

**Error** insufficient PHP version
error-php-insufficient

**Error** insufficient PHP and MySQL version
error-php-mysql-insufficient

**Error** config file permissions
error-config-file-permissions

**Error** empty table prefix
error-table-prefix-empty

**Error** preventative constant set
error-prevented-by-constant

**Error** already installed
error-already-installed

**Error** sample config file missing
error-missing-sample-config

There are more views, as well, such as when the user tries to add wonky characters to the table prefix... or uses unsupported characters in their email address. Those are also accounted for. I've just run short on time and need to get this posted.

### CODE REVIEW

**Comment by octoxan** [comment]
Looks fantastic!

File: src/wp-admin/install.php [hunk 1]
```diff
@@ -229,11 +66,13 @@ function display_setup_form( $error = null ) {
...
+// Get and check for sufficient versions of PHP and MySQL.
+$php_version   = phpversion();
+$mysql_version = $wpdb->db_version();
+$php_compat    = version_compare( $php_version, $required_php_version, '>=' );
+$mysql_compat  = version_compare( $mysql_version, $required_mysql_version, '>=' ) || file_exists( WP_CONTENT_DIR . '/db.php' );
 
+// Insufficient PHP and/or MySQL? Set a flag with an error message.
 if ( !$mysql_compat && !$php_compat ) {
 	/* translators: 1: ClassicPress version number, 2: Minimum required PHP version number, 3: Minimum required MySQL version number, 4: Current PHP version number, 5: Current MySQL version number */
 	$compat = sprintf( __( 'You cannot install because <a href="https://codex.wordpress.org/Version_%1$s">ClassicPress %1$s</a> requires PHP version %2$s or higher and MySQL version %3$s or higher. You are running PHP version %4$s and MySQL version %5$s.' ), $wp_version, $required_php_version, $required_mysql_version, $php_version, $mysql_version );
```
Comment (nylen) [inline src/wp-admin/install.php]: Let's update these strings too.  Right now they indicate the WordPress-equivalent version (currently 4.9.8) and link to the WP codex.

The logic for determining the link to the ClassicPress version can work as follows:

- If `classicpress_version()` does not contain a `+` character, link to `https://github.com/ClassicPress/ClassicPress-release/releases/%1$s` (e.g. https://github.com/ClassicPress/ClassicPress-release/releases/1.0.0-beta1).
- If `classicpress_version()` contains a `+` character, there are several possibilities for why, not all of which actually have an official release build.  In that case we should probably just link to the list at https://github.com/ClassicPress/ClassicPress-release instead.

File: src/wp-admin/install.php [hunk 2]
```diff
@@ -245,34 +84,36 @@ function display_setup_form( $error = null ) {
...
-	die(
-		'<h1>' . __( 'Configuration Error' ) . '</h1>' .
-		'<p>' . sprintf(
-			/* translators: %s: wp-config.php */
-			__( 'Your %s file has an empty database table prefix, which is not supported.' ),
-			'<code>wp-config.php</code>'
-		) . '</p></body></html>'
-	);
+	display_header( 'cp-installation-error' );
+	echo '<h1>'. __('Configuration Error') . '</h1>';
```
Comment (nylen) [inline src/wp-admin/install.php]: Can add spaces here `__( 'Configuration Error' )`.

File: src/wp-admin/install.php [hunk 3]
```diff
@@ -299,119 +147,286 @@ function display_setup_form( $error = null ) {
...
+			echo '<p>' . $wpdb->error->get_error_message() . '</p>';
+			display_footer();
+		}
+		// Add another script to the mix... Is this used?
 		$scripts_to_print[] = 'user-profile';
-
+		// Print the branded page header.
 		display_header();
-		// Fill in the data we gathered
+		// Get and pare submitted data.
```
Comment (nylen) [inline src/wp-admin/install.php]: `parse` instead of `pare`?

File: src/wp-admin/install.php [hunk 4]
```diff
@@ -299,119 +147,286 @@ function display_setup_form( $error = null ) {
...
-			$error = true;
+			$error = __( 'You must provide an email address.' );
 		} elseif ( ! is_email( $admin_email ) ) {
-			// TODO: poka-yoke
-			display_setup_form( __( 'Sorry, that isn&#8217;t a valid email address. Email addresses look like <code>username@example.com</code>.' ) );
-			$error = true;
+			$error = __( 'Sorry, that isn&#8217;t a valid email address. Email addresses look like <code>username@example.com</code>.' );
 		}
-
+		if ($error) {
```
Comment (nylen) [inline src/wp-admin/install.php]: Can add spaces here: `if ( $error )`.

File: src/wp-admin/install.php [hunk 5]
```diff
@@ -299,119 +147,286 @@ function display_setup_form( $error = null ) {
...
+		echo '	</tr>' . "\n";
+		// Weak password confirmation.
+		echo '	<tr class="pw-weak">' . "\n";
+		echo '		<th scope="row">' . __( 'Confirm Password' ) . '</th>' . "\n";
+		echo '		<td>' . "\n";
+		echo '			<label>' . "\n";
+		echo '			<input type="checkbox" name="pw_weak" class="pw-checkbox" />' . __( 'Confirm use of weak password' );
+		echo '			</label>' . "\n";
+		echo '		</td>' . "\n";
+		echo '	</tr>' . "\n";
```
Comment (nylen) [inline src/wp-admin/install.php]: I like this approach to PHP/HTML soup.  The `echo` statements and the indentation provide visual alignment in the source code, and the `"\n"` makes the HTML more readable after it gets to the browser.  It's often necessary to inspect and make sure this is working as intended, so I think it's worth including.

**Comment by nylen** [comment]
Excellent work @johnalarcon!  I will test this out more thoroughly, but for now I've left a few minor comments on the code.

The build failure is because a new image was added without being minified:

2018-12-08t11 47 51-05 00

If you can run `npm install` and `grunt precommit` that will minify the images.  Then you should be able to commit the changes.

This can also wait until we organize the images a bit better.

I will give this some more thought, but for now, one consideration is that we need to keep the filenames of the old WordPress logo images the same, so that plugins/themes/sites can reference them the same way.

This means we either need to fit new images into the existing naming scheme, or make copies of any new images.  I'm not sure which is the best way, but so far we have been fitting things in as best we can.

File: src/wp-admin/install.php (hunk 5 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: Great! I was dreading the possibility of a request to revert it back. I'm not a fan of a ton of echos, but the points you make are exactly why I took this approach. 👍

**Comment by johnalarcon** [comment]
> Excellent work @johnalarcon! I will test this out more thoroughly, but for now I've left a few minor comments on the code.

Thanks! I've addressed some of the comments and updated the PR accordingly. I'll get to the remaining items soon.

> The build failure is because a new image was added without being minified:
The irony...

> If you can run `npm install` and `grunt precommit` that will minify the images. Then you should be able to commit the changes.

Ok.

> This can also wait until we organize the images a bit better.

> I will give this some more thought, but for now, one consideration is that we need to keep the filenames of the old WordPress logo images the same, so that plugins/themes/sites can reference them the same way.

> This means we either need to fit new images into the existing naming scheme, or make copies of any new images. I'm not sure which is the best way, but so far we have been fitting things in as best we can.

I've created issue #266 with a proposal to get this in order.

File: src/wp-admin/install.php (hunk 1 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: 👍

File: src/wp-admin/install.php (hunk 2 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: Added.

File: src/wp-admin/install.php (hunk 3 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: Pare...to trim. I've reworded it for clarity.

File: src/wp-admin/install.php (hunk 4 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: Added.

File: src/wp-admin/install.php (hunk 1 above)
Comment (johnalarcon) [reply src/wp-admin/install.php]: Updated.

**Comment by nylen** [comment]
From https://www.classicpress.net/democracy/#democracy-exceptions:

> As you can imagine, if every decision required a Petition or Referendum, the community would be bogged down with an unmanageable bureaucratic process that would stifle progress. As a result, there are a number of processes and tasks that don’t need to go through the Petition or Referendum process. These include:
>
> Minor modifications to existing functionality or processes

I think removing the Alright Sparky step qualifies, and it's a clear improvement with (as far as I know) no backwards compatibility implications.

It's also already implemented, so even better.

The code looks good at a glance, but I agree this needs a thorough review.  I haven't forgotten about it, and we'll get to it as soon as we can.

**Comment by johnalarcon** [comment]
Thanks for the update.

> > Minor modifications to existing functionality or processes
> 
> I think removing the Alright Sparky step qualifies, and it's a clear improvement with (as far as I know) no backwards compatibility implications.

Great!

> The code looks good at a glance, but I agree this needs a thorough review. I haven't forgotten about it, and we'll get to it as soon as we can.

No worries.

File: src/wp-admin/images/cp-logo-gradient.svg [hunk 6]
```diff
@@ -0,0 +1 @@
...5,1.32,4,4,0,0,1,1,1.56,6.06,6.06,0,0,1,.27,1.86,4.63,4.63,0,0,1-1.15,3.19,5.55,5.55,0,0,1-2.58,1.73,11.79,11.79,0,0,1-3.42.44,12.88,12.88,0,0,1-4.65-.75A8.85,8.85,0,0,1,155.07,33" style="fill:#017f98"/><path d="M170.58,33c-.38-.34-.45-.66-.24-.95l.82-1.11c.27-.35.67-.33,1.19.05a9.25,9.25,0,0,0,2.39,1.27,8.76,8.76,0,0,0,3,.48,6.11,6.11,0,0,0,1.79-.24,3,3,0,0,0,1.37-.89,2.46,2.46,0,0,0,.59-1.74,2.58,2.58,0,0,0-.28-1.2,2.17,2.17,0,0,0-.87-.91,6.64,6.64,0,0,0-1.16-.54,18.6,18.6,0,0,0-1.85-.56c-.61-.14-1.41-.37-2.41-.68A7.66,7.66,0,0,1,173,25.1a5.29,5.29,0,0,1-1.56-1.5,4.53,4.53,0,0,1-.66-2.62A4.3,4.3,0,0,1,171.89,18a5.32,5.32,0,0,1,2.7-1.68,11.34,11.34,0,0,1,3.14-.45,10.74,10.74,0,0,1,3.64.5,10.29,10.29,0,0,1,2.37,1.19.57.57,0,0,1,.17.93l-.82,1.06c-.26.34-.65.33-1.19,0a7.72,7.72,0,0,0-1.91-.86,6.69,6.69,0,0,0-4.13-.16,2.69,2.69,0,0,0-1.33.85,2.08,2.08,0,0,0-.51,1.4,2.88,2.88,0,0,0,.2,1.1,2.06,2.06,0,0,0,.75.89,4.21,4.21,0,0,0,1.53.66l2.79.7a14.28,14.28,0,0,1,2.29.77,6.92,6.92,0,0,1,2,1.32,4.18,4.18,0,0,1,1,1.56,6.06,6.06,0,0,1,.27,1.86,4.64,4.64,0,0,1-1.16,3.19,5.57,5.57,0,0,1-2.57,1.73,11.81,11.81,0,0,1-3.43.44,13,13,0,0,1-4.65-.75,8.9,8.9,0,0,1-2.5-1.28" style="fill:#017f98"/></svg>
```
Comment (nylen) [inline src/wp-admin/images/cp-logo-gradient.svg]: Please use this SVG file directly and remove the `.png` added here (it's very low-resolution anyway).  Fortunately, we are past the point where we need to care about browsers which can't handle SVG images, and we've already added several others.

Comment (johnalarcon) [reply src/wp-admin/images/cp-logo-gradient.svg]: I've updated the style sheet to use the direct URL for the logo image, but am unable to delete the 2 images from the pull request...I get a non-working delete button for the svg, and no delete button for the png.

Comment (nylen) [reply src/wp-admin/images/cp-logo-gradient.svg]: The GitHub web UI doesn't support this, I guess.  I can take care of it.

Comment (johnalarcon) [reply src/wp-admin/images/cp-logo-gradient.svg]: Thanks. I'll let you mark this as resolved when ready.

Comment (nylen) [reply src/wp-admin/images/cp-logo-gradient.svg]: Done.  Note, I updated the stylesheet to use the copy of the SVG from the core code instead of a remote URL.

**Comment by johnalarcon** [comment]
After reviewing Beta2, I notice the initial step of the installation – the language picker – is showing a lot of white space due to there being fewer languages. I think there were 11 of them. It's not clear when there will be additional languages added, so I was thinking it should be reduced in height to fit, so it doesn't give the impression that something is missing. What say you?

**Comment by nylen** [comment]
> I was thinking [the language picker in the installation] should be reduced in height to fit, so it doesn't give the impression that something is missing. What say you?

Sounds good to me.

**Comment by johnalarcon** [comment]
I changed the size a few days back... forgot to drop a comment.

File: src/wp-admin/install.php [hunk 7]
```diff
@@ -299,119 +170,286 @@ function display_setup_form( $error = null ) {
...
 		if ( $error === false ) {
 			$wpdb->show_errors();
 			$result = wp_install( $weblog_title, $user_name, $admin_email, $public, '', wp_slash( $admin_password ), $loaded_language );
-?>
+			echo '<h1>' . __( 'Installation Complete' ) . '</h1>' . "\n";
+			echo '<p>' . __( 'Your ClassicPress site is ready to go!' ) . '</p>' . "\n";
+			echo '<table class="form-table install-success">' . "\n";
+			echo '<tr>' . "\n";
+			echo '<th>' . __( 'Username' ) . '</th>' . "\n";
+			echo '<td><code>' . esc_html( sanitize_user( $user_name, true ) ) . '</code></td>' . "\n";
```
Comment (invisnet) [inline src/wp-admin/install.php]: Why do we need to sanitize the user at this point?

Comment (nylen) [reply src/wp-admin/install.php]: We've already verified that the user is sanitized.  I've removed this call.

**Comment by nylen** [comment]
Thanks for doing this @johnalarcon, it took a while but we are finishing it.

There were more button styles and colors here than you might expect, so I've modified them to look like the buttons in #223:

2019-02-19t02 27 07-05 00

This darker color scheme also has a better contrast ratio, and I've added some fixes for disabled buttons and other edge cases.

I was able to test this including all the progress / error / secondary screens.  Tomorrow I'll merge #366 first and then update this PR to use the new filename.

**Comment by johnalarcon** [comment]
Sounds good; thanks for the update.
